File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/lock/realm/DeleteLockUseCase.java
package com.lattis.ellipse.domain.interactor.lock.realm;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.LockRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 7/25/17.
 */

public class DeleteLockUseCase extends UseCase<Boolean>

{

    private LockRepository lockRepository;

    @Inject
    public DeleteLockUseCase(ThreadExecutor threadExecutor,
                           PostExecutionThread postExecutionThread,
                           LockRepository lockRepository) {
        super(threadExecutor, postExecutionThread);
        this.lockRepository = lockRepository;
    }


    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return this.lockRepository.deleteLock();
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/lock/realm/DeleteLockUseCase.kt
package com.lattis.domain.usecase.lock.realm

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.LockRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class DeleteLockUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val lockRepository: LockRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    override fun buildUseCaseObservable(): Observable<Boolean> {
        return lockRepository.deleteLock()
    }

}